.github/scripts/generate_workflow_names.py

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python
+
+import os
+import yaml
+
+rootdir = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'workflows')
+for subdir, _, files in os.walk(rootdir):
+    for file in files:
+        filepath = os.path.join(subdir, file)
+        if filepath.endswith(('.yml', '.yaml')):
+            with open(filepath) as f:
+                data = yaml.safe_load(f)
+                assert 'name' in data, 'Every GHA workflow must have a name.'
+                # The reason the below key is True is because yaml parses 'on:' as a boolean.
+                # Since the solutions to maintain the key as 'on' seems more complicated than it's worth,
+                # I decided to stick with having True as a key instead of hacking my way through YAML
+                # implementation that is bound to change and become better anyway. More context below:
+                # https://stackoverflow.com/questions/36463531/pyyaml-automatically-converting-certain-keys-to-boolean-values
+                if 'pull_request' in data[True]:
+                    print(data['name'])
```

Comment:
may I suggest a nice [`pathlib`]() for you in these trying times?
```suggestion
import yaml
from pathlib import Path

rootdir = Path(__file__).parent.parent / 'workflows'
for filepath in rootdir.iterdir():
    if filepath.suffix in {'.yml', '.yaml'}:
        with open(filepath) as f:
            data = yaml.safe_load(f)
            assert 'name' in data, 'Every GHA workflow must have a name.'
            # The reason the below key is True is because yaml parses 'on:' as a boolean.
            # Since the solutions to maintain the key as 'on' seems more complicated than it's worth,
            # I decided to stick with having True as a key instead of hacking my way through YAML
            # implementation that is bound to change and become better anyway. More context below:
            # https://stackoverflow.com/questions/36463531/pyyaml-automatically-converting-certain-keys-to-boolean-values
            if 'pull_request' in data[True]:
                print(data['name'])
```

Reply:
Is it possible for a GHA workflow to be NOT in the direct `.github/workflows` dir? I had a recursive solution but this could work in a shorter way

Reply:
I don't believe so, but maybe?

Reply:
You must store workflow files in the .github/workflows directory of your repository, says https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions, so I think this is good